crates/bevy_egui/src/systems.rs

```diff
@@ -0,0 +1,339 @@
+        });
+
+    let events = keyboard_input_reader.iter(&keyboard_input_events)
+        .fold(Vec::<egui::Event>::new(), |mut events: Vec<egui::Event>, KeyboardInput { key_code, state, .. }| {
+            if let Some(key_code) = key_code {
+                // Handle special cased key combos like cut/copy/paste
+                if let Some(event) = match key_code {
+                    KeyCode::Copy => Some(Event::Copy),
+                    KeyCode::Cut => Some(Event::Cut),
+                    KeyCode::Paste => {
+                        tracing::warn!("Paste event isn't implemented in bevy_egui! We need to figure out how to get stuff from the clipboard...");
+                        None
+                    }
+                    _ => None
+                } {
+                    events.push(event);
+                } else if let Some(key) = key_code.into_egui_key() {
+                    events.push(Event::Key {
+                        key,
+                        pressed: state.is_pressed()
+                    });
+                } else {
+                    tracing::info!("Key presses are currently not really given to egui! Pressed: {:?}", key_code);
+                }
+            }
+
+            events
+        });
+
+    let new_input = egui::RawInput {
+        mouse_down,
+        mouse_pos,
+        scroll_delta,
+        screen_size,
+        events,
+        ..*previous_input
+    };
+
+    *previous_input = new_input.clone();
+    egui_input.raw_input = new_input;
+}
+
+trait IntoEguiKey {
+    fn into_egui_key(self) -> Option<egui::Key>;
+}
+
+impl IntoEguiKey for KeyCode {
+    fn into_egui_key(self) -> Option<egui::Key> {
+        let key = match self {
+            KeyCode::LAlt |
+                KeyCode::RAlt =>    egui::Key::Alt,
+            KeyCode::Back =>        egui::Key::Backspace,
+            KeyCode::LControl |
+                KeyCode::RControl =>egui::Key::Control,
+            KeyCode::Delete =>      egui::Key::Delete,
+            KeyCode::Down =>        egui::Key::Down,
+            KeyCode::End =>         egui::Key::End,
+            KeyCode::Escape =>      egui::Key::Escape,
+            KeyCode::Home =>        egui::Key::Home,
+            KeyCode::Insert =>      egui::Key::Insert,
+            KeyCode::Left =>        egui::Key::Left,
+            KeyCode::LWin |
+                KeyCode::RWin =>    egui::Key::Logo,
+            KeyCode::PageDown =>    egui::Key::PageDown,
+            KeyCode::PageUp =>      egui::Key::PageUp,
+            KeyCode::NumpadEnter |
+                KeyCode:: Return => egui::Key::Enter,
+            KeyCode::Right =>       egui::Key::Right,
+            KeyCode::LShift |
+                KeyCode::RShift =>  egui::Key::Shift,
+            KeyCode::Tab =>         egui::Key::Tab,
+            KeyCode::Up =>          egui::Key::Up,
+            
+            _ => return None
+        };
+
+        Some(key)
```

Comment:
```suggestion
        Some(match self {
            KeyCode::LAlt |
                KeyCode::RAlt =>    egui::Key::Alt,
            KeyCode::Back =>        egui::Key::Backspace,
            KeyCode::LControl |
                KeyCode::RControl =>egui::Key::Control,
            KeyCode::Delete =>      egui::Key::Delete,
            KeyCode::Down =>        egui::Key::Down,
            KeyCode::End =>         egui::Key::End,
            KeyCode::Escape =>      egui::Key::Escape,
            KeyCode::Home =>        egui::Key::Home,
            KeyCode::Insert =>      egui::Key::Insert,
            KeyCode::Left =>        egui::Key::Left,
            KeyCode::LWin |
                KeyCode::RWin =>    egui::Key::Logo,
            KeyCode::PageDown =>    egui::Key::PageDown,
            KeyCode::PageUp =>      egui::Key::PageUp,
            KeyCode::NumpadEnter |
                KeyCode:: Return => egui::Key::Enter,
            KeyCode::Right =>       egui::Key::Right,
            KeyCode::LShift |
                KeyCode::RShift =>  egui::Key::Shift,
            KeyCode::Tab =>         egui::Key::Tab,
            KeyCode::Up =>          egui::Key::Up,
            
            _ => return None
        })
```
If you're gonna be lazy with it, may as well just inline it.

Reply:
Well like... I also understand this one, but I think it's harder to tell what's happening. It makes the `Some` part appear at the start, versus at the end, which I find harder to read personally.